synapse/config/redis.py

```diff
@@ -54,4 +60,16 @@ def generate_config_section(self, **kwargs: Any) -> str:
           # Optional password if configured on the Redis instance
           #
           #password: <secret_password>
+
+          # Optional one or more Redis hosts to use for long term sharedd caches.
+          # Should be configured to automatically expire records when out of
+          # memory, and not be the same instance as used for replication.
+          #
+          #cache_shards:
+          #  enabled: false
+          #  expire_caches: false
```

Comment:
Not currently supported but could be fairly easily using a Redis transaction block and setting each key one by one.